packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -0,0 +1,277 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../android/android_sdk.dart';
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../base/utils.dart';
+import '../base/version.dart';
+import '../build_info.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../project.dart';
+import '../reporting/reporting.dart';
+import 'android_sdk.dart';
+import 'android_studio.dart';
+
+/// The environment variables needed to run Gradle.
+Map<String, String> get gradleEnv {
+  final Map<String, String> env = Map<String, String>.from(platform.environment);
+  if (javaPath != null) {
+    // Use java bundled with Android Studio.
+    env['JAVA_HOME'] = javaPath;
+  }
+  // Don't log analytics for downstream Flutter commands.
+  // e.g. `flutter build bundle`.
+  env['FLUTTER_SUPPRESS_ANALYTICS'] = 'true';
```

Comment:
it would be cleaner if flutter was always called with an argument that did that, rather than relying on the environment variables.

Reply:
+1.  When the tool calls Gradle, it doesn't log analytics. However, in add2app scenarios, when a developer calls Gradle directly (from Android Studio, for example), it's preferred to log the calls to the tool.